### ISSUE TITLE
Element 11

### ISSUE THREAD

**Issue by ClimberMel** [ClimberMel/SMF_Add-in#37]
I am adding all the smfGetYahooPortfolioView Items into smf-elements-1.txt to use with RCHGetElementNumber

Here are elements 11 & 16 for examples
11;YahooQT;GMT Offset Milliseconds;=smfGetYahooJSONData("~~~~~","quoteType","gmtOffSetMilliseconds",,"")
16;YahooPrice;Last Traded (UNIX);=smfGetYahooJSONData("~~~~~","price","regularMarketTime",,"")

The JSON comes from

Here is the JSON:
![image](https://github.com/ClimberMel/SMF_Add-in/assets/9968993/01fc46d8-aeb5-4843-b36d-f9d5094488f8)

Instead of getting the value for key gmtOffSetMilliseconds, it is getting the next item in quotes which is maxAge

I also tried using JsonConverter, but it fails on that as well...

**Comment by Bruce-95** [comment]
Mel,

I tested some code changes to the smfGetYahooJSONData() function in module modGetYahooJSONData and it appears to fix Issue #37. In the area where the "Type" parameter is evaluated in smfGetYahooJSONData(), I've add a CASE stmt.

...
Select Case LCase(pProcess)
      Case "": s1 = smfStrExtr(s1, """", """")
      Case "fmt": s1 = smfStrExtr(s1, """fmt"":""", """")
      Case "raw": s1 = smfStrExtr(s1, """raw"":", ",")
      Case "num": s1 = smfStrExtr(s1, "~", ",""")                                      ' <== fix for Issue #37
      Case Else: s1 = smfStrExtr(s1, """raw"":", ",")
      End Select
...

This requires a new Type parameter of "num" when calling the function for certain Field_Names. This is what I got for APPL on 06/03/23 11:35am.

=smfGetYahooJSONData("AAPL", "quoteType", "gmtOffSetMilliseconds", , "num") ........ returns -14400000
=smfGetYahooJSONData("AAPL", "quoteType","firstTradeDateEpochUtc",,"num") .......... returns 345479400
=smfGetYahooJSONData("AAPL","price","regularMarketTime",,"num") ................................ returns 1685736005

These match the data in the "APPL" JSON file(s). I have not done an exhaustive test for other date/time fields. It seems you'll need the "num" Type when extracting any date/time fields or numeric fields that are not prefixed with "raw", "fmt" or are not quoted.

As for "smf-elements-1.txt" , you would add "num" to the following:
...
11;YahooQT;GMT Offset Milliseconds;=smfGetYahooJSONData("~~~~~","quoteType","gmtOffSetMilliseconds",,"num")

16;YahooPrice;Last Traded (UNIX);=smfGetYahooJSONData("~~~~~","price","regularMarketTime",,"num")
...

Running =RCHGetElementNumber("AAPL",11) and =RCHGetElementNumber("AAPL",16) work as well with the new entries.

As I have never used Github (even this is my 1st time making a comment) I will leave it to you to make the changes on a date of your choosing. 

To summarize, 
- update smfGetYahooJSONData() function code and comments;
- update "smf-elements-1.txt";
- update smfGetYahooJSONData doc - (see attached capture2.png)
- update ChangeLog doc - (see attached capture.png)
- release new zip file.

![Capture2](https://github.com/ClimberMel/SMF_Add-in/assets/135463214/0048f139-e9ef-400e-8e11-63d6b929bc20)

![Capture](https://github.com/ClimberMel/SMF_Add-in/assets/135463214/a08d595c-2f4f-4208-9367-685f290d2394)


Will try to answer any question,
~ Bruce

**Comment by ClimberMel** [comment]
Thanks Bruce!  I'm making tomorrow an office day, so I'll check out bot your code and the html.  I tend to just write html, plus I have css to take care of most of the formatting and keep the pages consistent.  But I'm sure it will save me a lot of time.

Mel

**Comment by ClimberMel** [comment]
I'm adding those changes to the functions and hope to have a new version out today or tomorrow.  I am including updates to the elements files and I will update the documentation soon.
